python/paddle/fluid/layers/control_flow.py

```diff
@@ -2785,6 +2786,165 @@ def _assert_in_rnn_block_(self, method):
+        TypeError: If the type of ``branch_fns`` is not dict, list or tuple.
+        TypeError: If the elements of ``branch_fns`` is not 2-tuple.
+        TypeError: If the first element of 2-tuple in ``branch_fns`` is not integer.
+        ValueError: If the first element of 2-tuple in ``branch_fns`` is not unique.
+        TypeError: If the second element of 2-tuple in ``branch_fns`` is not callable.
+        TypeError: If ``default`` is not None but it is not callable.
+
+    Examples:
+        .. code-block:: python
+
+            import paddle.fluid as fluid
+            def fn_1():
+                return layers.fill_constant(shape=[1, 2], dtype='float32', value=1)
+
+            def fn_2():
+                return layers.fill_constant(shape=[2, 2], dtype='int32', value=2)
+
+            def fn_3():
+                return layers.fill_constant(shape=[3], dtype='int32', value=3)
+
+            main_program = fluid.default_startup_program()
+            startup_program = fluid.default_main_program()
+            with program_guard(main_program, startup_program):
+                index_1 = layers.fill_constant(shape=[1], dtype='int32', value=1)
+                index_2 = layers.fill_constant(shape=[1], dtype='int32', value=2)
+
+                out_1 = layers.switch_case(
+                    branch_index=index_1,
+                    branch_fns={1: fn_1, 2: fn_2},
+                    default=fn_3)
+
+                out_2 = layers.switch_case(
+                    branch_index=index_2,
+                    branch_fns=[(1, fn_1), (2, fn_2)],
+                    default=fn_3)
+
+                # Argument default is None and no index matches. fn_3 will be called because of the max index 7.
+                out_3 = layers.switch_case(
+                    branch_index=index_2,
+                    branch_fns=[(0, fn_1), (4, fn_2), (7, fn_3)])
+
+                exe = fluid.Executor(fluid.CPUPlace())
+                res_1, res_2, res_3 = exe.run(main_program,
+                                              fetch_list=[out_1, out_2, out_3])
+                print(res_1)  # [[1. 1.]]
+                print(res_2)  # [[2 2] [2 2]]
+                print(res_3)  # [3 3 3]
+    '''
+    helper = LayerHelper('switch_case', **locals())
+
+    def _check_args(branch_index, branch_fns, default):
+        def _error_message(what, arg_name, op_name, right_value, error_value):
+            raise TypeError(
+                "{what} of '{arg_name}' in Op({op_name}) "
+                "must be {right_value}, but received: {error_value}.".format(
+                    what=what,
+                    arg_name=arg_name,
+                    op_name=op_name,
+                    right_value=right_value,
+                    error_value=error_value))
+
+        if not isinstance(branch_index, Variable):
+            raise TypeError(
+                _error_message("The type", "branch_index", "switch_case",
```

Comment:
I saw you raise TypeError in _error_message and also raise here, that's not good. 

Two options:
1. Don't raise TypeError in _error_message, but return the error message as a string in _error_message

2. Remove "raise TypeError" here, rename "_error_message" to raise_error_with_message or some other name properly.